packages/thyra/core/src/interfaces/operator_solve/fundamental/Thyra_LinearOpWithSolveBase_decl.hpp

```diff
@@ -279,6 +279,8 @@ class LinearOpWithSolveBase
 {
 public:
 
+  using scalar_type = Scalar;
```

Comment:
Just curious, but why was this typedef needed?

Reply:
I needed it for this:
https://github.com/trilinos/Trilinos/blob/9aedfd1ac818ee731de8db82e8ead41eaa590dc4/packages/PyTrilinos2/src/PyTrilinos2_Thyra_Custom.hpp#L6-L15
The problem is that I currently cannot seem to get pybind11 to accept both `Teuchos::RCP` and `Teuchos::Ptr` to be interpreted as custom smart pointers. So I had to manually add a `solve` method that takes an `RCP` instead of `Ptr`. In order to do this, I have to extract `Scalar` out of `Thyra::LinearOpWithSolveBase<Scalar>`.